release-26.2.1-rc: pkg/sql/logictest/tests/fakedist/fakedist_test: TestLogic_row_level_ttl failed

pkg/sql/logictest/tests/fakedist/fakedist_test.TestLogic_row_level_ttl [failed](https://mesolite.cluster.engflow.com/invocations/default/86dbb7ac-a646-4eb1-a2ea-0ca9a1227238?testReportRun=2&testReportShard=44&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9mYWtlZGlzdDpmYWtlZGlzdF90ZXN0) on release-26.2.1-rc @ [b398a6e1fde74a915daf8e9841303c8ee39c2abe](https://github.com/cockroachdb/cockroach/commits/b398a6e1fde74a915daf8e9841303c8ee39c2abe):


```
=== RUN   TestLogic_row_level_ttl
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestLogic_row_level_ttl2680584923
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    test_server_shim.go:150: cluster virtualization disabled due to issue: #156124 (expected label: C-bug)
[11:14:52] setting distsql_workmem='38232B';
[11:16:49] --- done: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/row_level_ttl with config fakedist: 251 tests, 2 failures
[11:17:15] --- total progress: 251 statements
--- total: 251 tests, 2 failures
    logic.go:4871: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_row_level_ttl2680584923
--- FAIL: TestLogic_row_level_ttl (150.20s)
=== RUN   TestLogic_row_level_ttl/execute_schedule
    logic.go:3327: let $schedule_id_1 = 1179026089202221057
    logic.go:3327: let $schedule_id_2 = 1179026090519527425
    logic.go:3277: 
         
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/row_level_ttl:1838: SELECT count(*) FROM [SHOW SCHEDULES] WHERE label LIKE '%tbl_execute_schedule%' AND next_run <= now()
        expected:
            2
            
        but found (query options: "") :
            1
            
    logic.go:2433: 
         /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/row_level_ttl:1843: too many errors encountered, skipping the rest of the input
--- FAIL: TestLogic_row_level_ttl/execute_schedule (1.38s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=44</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_row_level_ttl.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-64275

--- a/pkg/sql/logictest/testdata/logic_test/row_level_ttl
+++ b/pkg/sql/logictest/testdata/logic_test/row_level_ttl
@@ -1777,6 +1777,15 @@ subtest end
 
 subtest execute_schedule
 
+# Disable the job scheduler daemon for this subtest. Otherwise, the daemon can
+# scan system.scheduled_jobs in the background, find a schedule that EXECUTE
+# SCHEDULE just made eligible (next_run <= now()), run it, and reschedule
+# next_run to the future (tomorrow, for @daily). That would race with the
+# assertions below that check next_run remains in the past, causing flakes.
+# See https://github.com/cockroachdb/cockroach/issues/170997.
+statement ok
+SET CLUSTER SETTING jobs.scheduler.enabled = false
+
 statement ok
 CREATE TABLE tbl_execute_schedule (
   id INT PRIMARY KEY,
@@ -1848,5 +1857,9 @@ EXECUTE SCHEDULE $schedule_id_1
 
 statement ok
 RESUME SCHEDULE $schedule_id_1
+
+# Re-enable the job scheduler daemon disabled at the start of this subtest.
+statement ok
+RESET CLUSTER SETTING jobs.scheduler.enabled
 
 subtest end

